.github/workflows/deploy-book-ghpages.yml

```diff
@@ -27,15 +41,21 @@ jobs:
     - uses: actions/checkout@v4
       with:
         fetch-depth: 0
+
     - id: set-branches
-      name: Collect all remote branch names for the matrix
+      name: Set branches
       run: |
-        branches=$(git branch -r | sed 's,\s*origin/,,g' | jq -Rn '[inputs]')
+        if [ "$BRANCHES_TO_DEPLOY" == '*' ]; then
+          branches=$(git branch -r | sed 's,\s*origin/,,g' | jq -Rn '[inputs]')
```

Comment:
What's the purpose of this? Piping to `jq` is a bit unclear here, given that `git branch` doesn't output JSON

Reply:
Here `jq` just takes the raw (`-R`) STDIN lines & makes a JSON list out of them (since the matrix variable `branch` has to be JSON)